ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivDistributeRawTransaction.java

```diff
@@ -88,14 +87,30 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
         return new JsonRpcErrorResponse(id, JsonRpcError.ONCHAIN_PRIVACY_GROUP_ID_NOT_AVAILABLE);
       }
 
-      final Optional<PrivacyGroup> maybePrivacyGroup =
-          onchainPrivacyGroupsEnabled
-              ? findOnchainPrivacyGroup(
-                  privacyController, maybePrivacyGroupId, privacyUserId, privateTransaction)
-              : findOffchainPrivacyGroup(privacyController, maybePrivacyGroupId, privacyUserId);
-
-      if (onchainPrivacyGroupsEnabled && maybePrivacyGroup.isEmpty()) {
-        return new JsonRpcErrorResponse(id, JsonRpcError.ONCHAIN_PRIVACY_GROUP_DOES_NOT_EXIST);
+      Optional<PrivacyGroup> maybePrivacyGroup =
+          maybePrivacyGroupId.flatMap(
+              gId ->
+                  privacyController.findPrivacyGroupByGroupId(gId.toBase64String(), privacyUserId));
+
+      if (onchainPrivacyGroupsEnabled) {
+        if (OnchainUtil.isGroupAdditionTransaction(privateTransaction)) {
+          final List<String> participantsFromParameter =
+              OnchainUtil.getParticipantsFromParameter(privateTransaction.getPayload());
+          if (maybePrivacyGroup.isEmpty()) {
+            maybePrivacyGroup =
+                Optional.of(
+                    new PrivacyGroup(
```

Comment:
Why do we have this case (privacy group not present and we have to create it here) now when we did not have it before?

Reply:
There was onchain specific code in the restricted controller. This did not really fit into the controller interface, so I moved it here ... 
And I think there is something similar in the eea send raw transaction class.